torch/_functorch/_aot_autograd/utils.py

```diff
@@ -117,10 +117,10 @@ def call_func_at_runtime_with_args(f, args, steal_args=False, disable_amp=False)
 class PytreeThunk:
     spec: Optional[pytree.TreeSpec] = None
     # These are some kinda dumb microoptimizations that save about 3-4 us of overhead.
-    is_simple = (
-        None  # if the output spec is a tuple/list, we won't bother unflattening it.
-    )
-    is_really_simple = None  # if the output spec is a LeafSpec
+    is_simple: Optional[
```

Comment:
I have a ruff rule for eradicating implicit optionals, but I couldn't figure out how to get all the backwards compatibility tests passing. `RUF013` if you want to give it a try. Also has autofixes.

Reply:
For what it's worth, implicit optionals isn't really a priority for me right now, I want to target bigger semantic confusion in the codebase first lol